tls/conn.go

```diff
@@ -626,7 +628,13 @@ Again:
 			c.in.freeBlock(b)
 			goto Again
 		case alertLevelError:
-			c.in.setErrorLocked(&net.OpError{Op: "remote error", Err: alert(data[1])})
+			var e error
+			if alert(data[1]) == alertHandshakeFailure {
```

Comment:
This if statement doesn't actually change the behavior of the program, so I don't think it needs to be added here. I would very much prefer not to modify original `crypto/tls` library code as it will make merging changes from upstream more complicated.